Perl has deprecated the feature:

    defined(@array)

In Perl 5.22+, this deprecation warning has been converted
to a fatal error. This results in the following error message:

    Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at milestone.pl line 58.

This patch takes the suggestion of omitting defined().
--- a/js/src/config/milestone.pl	2017-02-14 15:00:42.234275243 -0600
+++ b/js/src/config/milestone.pl	2017-02-14 15:01:19.453313360 -0600
@@ -55,7 +55,7 @@
 #
 my $milestone = Moz::Milestone::getOfficialMilestone($MILESTONE_FILE);
 
-if (defined(@TEMPLATE_FILE)) {
+if (@TEMPLATE_FILE) {
   my $TFILE;
 
   foreach $TFILE (@TEMPLATE_FILE) {
